src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx

```diff
@@ -98,10 +101,12 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
   }
 
   const handleDeleted = () => {
-    trackEvent({
-      action: ActionType.deletedSavedSearch,
-      saved_search_id: editAlertEntity?.id,
-    })
+    if (editAlertEntity) {
+      trackEvent({
+        action: ActionType.deletedSavedSearch,
+        saved_search_id: editAlertEntity.id,
+      })
+    }
```

Comment:
I wonder if this check is really needed 🤔 If it's just about `editAlertEntity.id` we could change it to an early return (because logically we don't want to do anything in this method if `editAlertEntity` isn't present or add an `?`:

```suggestion
    if (!editAlertEntity) return

    trackEvent({
      action: ActionType.deletedSavedSearch,
      saved_search_id: editAlertEntity.id,
    })
```

---

```suggestion
      trackEvent({
        action: ActionType.deletedSavedSearch,
        saved_search_id: editAlertEntity?.id,
      })
```